src/observables/OptionStream.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Options } from "../omnisharp/options";
+import { vscode } from "../vscodeAdapter";
+import 'rxjs/add/operator/take';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import Disposable, { IDisposable } from "../Disposable";
+import { Subject } from "rxjs/Subject";
+
+export default class OptionStream {
+    private optionStream: Subject<Options>;
+    private disposable: IDisposable;
+
+    constructor(vscode: vscode) {
+        this.optionStream = new BehaviorSubject<Options>(Options.Read(vscode));
```

Comment:
I thought you were making this into a regular Subject?

Reply:
An subject wont retain the most recent value for the observers that subscribe to it after the value is pushed into the subject. Hence we need to use the behavior subject here.